auth_token/models.py

```
@@ -40,16 +42,22 @@ def _hash_key(self, key, **kwargs):
         return hash_key(key)
 
     def create(self, key_generator, **kwargs):
-        for attempt in range(settings.MAX_RANDOM_KEY_ITERATIONS + 1):
+        for attempt in range(1, settings.MAX_RANDOM_KEY_ITERATIONS + 1):
             try:
                 key = key_generator()
                 hashed_key = self._hash_key(key, **kwargs)
                 obj = super().create(key=hashed_key, **kwargs)
                 obj.secret_key = key
                 return obj
             except IntegrityError:
-                if attempt > settings.MAX_RANDOM_KEY_ITERATIONS:
-                    raise IntegrityError('Could not produce unique key')
+                if attempt >= settings.MAX_RANDOM_KEY_ITERATIONS:
+                    raise KeyGeneratorError('Could not produce unique key')
+            except PersistenceException as ex:
+                if ex.error_dict and 'key' in ex.error_dict:
```

only if bug is returned for key field. The new key is generated.

---

Because I didn't want to hide other errors